src/saltext/vmware/utils/license_mgr.py

```diff
@@ -0,0 +1,399 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=no-name-in-module
+import logging
+
+# noreorder
+import salt.exceptions
+import salt.modules.cmdmod
+import salt.utils.path
+import salt.utils.platform
+import salt.utils.stringutils
+import saltext.vmware.utils.cluster as utils_cluster
+import saltext.vmware.utils.datacenter as utils_datacenter
+import saltext.vmware.utils.esxi as utils_esxi
+from saltext.vmware.utils.common import get_service_content as get_service_content
+
+try:
+    from pyVmomi import vim, vmodl
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+log = logging.getLogger(__name__)
+
+
+def _list_lic_keys(licenses):
+    """
+    Return a list of license keys
+
+    licenses:
+       vCenter licenses as made available from licenseManager
+
+    Note:
+        skips evaluation licenses
+    """
+    lic_keys = []
+    for lic in licenses:
+        if lic.used is None:
+            continue
+        lic_keys.append(lic.licenseKey)
+    return lic_keys
+
+
+def _find_lic_for_key(licenses, license_key):
+    """
+    Return license which matches specified license_key
+
+    licenses:
+       vCenter licenses as made available from licenseManager
+
+    license_key
+        license key to search for
+
+    Note:
+        skips evaluation licenses
+    """
+    lic_keys = []
+    for lic in licenses:
+        if lic.used is None:
+            continue
+        if license_key == lic.licenseKey:
+            return lic
+    return None
+
+
+def _get_entity_id(service_instance, datacenter_name, cluster_name, esxi_hostname):
+    """
+    Get entity_id for datacenter, cluster, ESXi Server or vCenter
+
+    If no datacenter, cluster or ESXi Server is specified, it is assumed the operation is to be applied to a vCenter
+
+    service_instance:
+        Service Instance
+
+    datacenter_name
+        Datacenter name to use for the operation
+
+    cluster_name
+        Name of the cluster to use for the operation
+
+    esxi_hostname
+        Hostname of the ESXi Server use for the operation
+    """
+
+    # need to extract entity this license is intended for
+    # from choices:
+    #   1 vCenter - _entity_id = service_instance.content.about.instanceUuid
+    #   2 ESXi - _entity_id = esxi host _moId
+    #   3 cluster - _entity_id = cluster _moId
+
+    _entity_id = None  # TBD miracle happens here and have value
+    datacenter_ref = None
+    if not any(datacenter_name or cluster_name or esxi_hostname):
```

Comment:
**suggestion** `any` takes an iterable - either a tuple or a list would be fine. Alternatively, drop the 'any'.
```suggestion
    if not any((datacenter_name, cluster_name, esxi_hostname)):
```   

(the other option is `not (datacenter_name or cluster_name or esxi_hostname)`)

Reply:
fixed